WebRoot/js/zimbraMail/briefcase/view/ZmPreviewPaneView.js

```diff
@@ -355,37 +356,44 @@ function() {
 	this._previewView.set(this._delayedSelectionItem);
 };
 
-ZmPreviewPaneView.prototype._listSelectionListener = function(ev, item) {
-
-    var item = item || ev.item;
-    if (!item) {
-    	return;
-    }
+ZmPreviewPaneView.prototype._listSelectionListener = function (ev, item) {
+	var item = item || ev.item;
+	if (!item) {
+		return;
+	}
 
-    var cs = appCtxt.isOffline && appCtxt.getCurrentSearch();
-    if (cs) {
-        appCtxt.accountList.setActiveAccount(item.getAccount());
-    }
-    var noChange = ev && ev._details && ev._details.oldFolderId == item.folderId;
-    // Ignore (no preview change) if move to same folder, deletion, or multi-select (shift key)
-    if ((ev.event === ZmEvent.E_MOVE && noChange) || ev.event === ZmEvent.E_DELETE || ev.shiftKey) {
-        return;
-    }
+	var cs = appCtxt.isOffline && appCtxt.getCurrentSearch();
+	if (cs) {
+		appCtxt.accountList.setActiveAccount(item.getAccount());
+	}
+	var noChange = ev && ev._details && ev._details.oldFolderId == item.folderId;
+	// Ignore (no preview change) if move to same folder, deletion, or multi-select (shift key)
+	if ((ev.event === ZmEvent.E_MOVE && noChange) || ev.event === ZmEvent.E_DELETE || ev.shiftKey) {
+		return;
+	}
 
-    if(ev.field == ZmItem.F_EXPAND && this._detailListView._isExpandable(item)){
-        this._detailListView.expandItem(item);   
-    } else if(this._controller.isReadingPaneOn() && item){
-    	if (ev.kbNavEvent) {
-    		if (this._listSelectionShortcutDelayActionId) {
-    			AjxTimedAction.cancelAction(this._listSelectionShortcutDelayActionId); 
-    		}
-    		this._delayedSelectionItem = item;
-    		this._listSelectionShortcutDelayActionId = AjxTimedAction.scheduleAction(this._listSelectionShortcutDelayAction,
-    				ZmPreviewPaneView.LIST_SELECTION_SHORTCUT_DELAY)
-    	} else {
-    		this._previewView.set(item);
-    	}
-    }
+	if (ev.field == ZmItem.F_EXPAND && this._detailListView._isExpandable(item)) {
+		this._detailListView.expandItem(item);
+	} else if (item) {
+		if (this._controller.isReadingPaneOn()) {
+			if (ev.kbNavEvent) {
+				if (this._listSelectionShortcutDelayActionId) {
+					AjxTimedAction.cancelAction(this._listSelectionShortcutDelayActionId);
+				}
+				this._delayedSelectionItem = item;
+				this._listSelectionShortcutDelayActionId = AjxTimedAction.scheduleAction(
+					this._listSelectionShortcutDelayAction,
+					ZmPreviewPaneView.LIST_SELECTION_SHORTCUT_DELAY
+				);
+			} else {
+				this._previewView.set(item);
+			}
+		} else if (item.folderId == ZmFolder.ID_FILE_SHARED_WITH_ME) {
+			var restUrl = item.getRestUrl();
+			restUrl = AjxStringUtil.fixCrossDomainReference(restUrl);
```

Comment:
not sure if this is really needed, this function is used to fix cross domain URLs when adding URL in iframe but here we are just sending XMLHTTPRequest
I think it's useful in _handlePreviewUrlResponse function not here

Reply:
applied the suggested change.